src/main/java/pl/stormit/eduquiz/authenticator/CustomUserDetailService.java

```diff
@@ -0,0 +1,32 @@
+package pl.stormit.eduquiz.authenticator;
+
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+import pl.stormit.eduquiz.quizcreator.domain.user.User;
+import pl.stormit.eduquiz.quizcreator.domain.user.UserRepository;
+
+import java.util.Collections;
+
+@Component
+public class CustomUserDetailService implements UserDetailsService {
+
+
+    private final UserRepository userRepository;
+
+    public CustomUserDetailService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String nickname) throws UsernameNotFoundException {
+        User user = userRepository.findByNickname(nickname)
+                .orElseThrow(() -> new UsernameNotFoundException("There is no such user"));
```

Comment:
We should use a message something like this:
"The user by nickname: " + nickname + ", does not exist."
This matter was discussed during a code review conducted by Tomek Woliński.